src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -111,7 +114,42 @@ public string GetHeaderValue(ReadOnlySpan<byte> headerValue)
                 }
             }
 
+            if (KnownHeader != null && KnownHeader.Name == KnownHeaders.Location.Name)
+            {
+                // Normally Location should be in ISO-8859-1, ocassionally some servers respond with UTF-8 though
+                if (TryDecodeUtf8(headerValue, out string decoded))
+                {
+                    return decoded;
+                }
+            }
+
             return HttpRuleParser.DefaultHttpEncoding.GetString(headerValue);
         }
+
+        private static bool TryDecodeUtf8(ReadOnlySpan<byte> input, out string decoded)
+        {
+            bool possibleUtf8 = false;
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (input[i] > 127)
+                {
+                    possibleUtf8 = true;
+                    break;
+                }
+            }
+
+            if (possibleUtf8)
+            {
+                try
+                {
+                    decoded = s_utf8DecoderWithExceptionFallback.GetString(input);
```

Comment:
>s_utf8DecoderWithExceptionFallback [](start = 30, length = 34)

please test it if you have high surrogate without low surrogate.

Reply:
When I pass incomplete: `0xf0, 0x90, 0x8d, 0x88` (i.e. `0xf0, 0x90, 0x8d`) I get DecoderFallbackException as expected